test/worldclim-future.jl

```diff
@@ -0,0 +1,41 @@
+using RasterDataSources, Test, Dates
+using RasterDataSources: rasterurl, rastername, rasterpath
+
+@testset "WorldClim Future BioClim CMIP6" begin
+    bioclim_name = "wc2.1_10m_bioc_MRI-ESM2-0_ssp126_2041-2060.tif"
+    @test rastername(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, 5; date=Date(2050), res = "10m") == bioclim_name
+    bioclim_path = joinpath(ENV["RASTERDATASOURCES_PATH"], "WorldClim", "Future", "BioClim", "ssp126", "MRI-ESM2-0")
+    @test rasterpath(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}) == bioclim_path
+
+    raster_path = joinpath(bioclim_path, bioclim_name)
+    raster_path2 = joinpath(bioclim_path, "wc2.1_10m_5_MRI-ESM2-0_SSP126_2041-2060.tif")
+    @test getraster(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, 5; date=Date(2050), res = "10m") == raster_path
+    @test getraster(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, :bio5; date=Date(2050)) == raster_path
+
+#=
+    @test getraster(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, (5,); date=Date(2050), res = "10m") == (bio5=raster_path,)
+    @test getraster(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, [5]; date=Date(2050), res = "10m") == (bio5=raster_path,)
+    @test getraster(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, (5,); date=[Date(2050)], res = "10m") == 
+        [(bio5=raster_path,)]
+    @test getraster(WorldClim{Future{BioClim,CMIP6,MRI_ESM2_0,SSP126}}, (:bio5,); date=Date(2050)) == (bio5=raster_path,)
+    =#
```

Comment:
I've adapted these tests from the CHELSA ones, and this is the only one that doesn't pass. WorldClim future has just 1 file for all bioclim variables, and currently `getraster` just returns the one filename after checking layers, regardless of how many layers are passed. I'm not entirely convinced that returning a namedtuple with the same filename repeated is the way to go.

Reply:
No I think returning a single filename is probably best, to force  consuming packages to deal with the fact it has all the bands in it.

I think for e.g. `Raster` and `RasterStack` we can add a hack to actually just select those bands so it feels normal to higher level users.

Reply:
Yes, one this is merged I'll make a PR to fix RastersRasterDataSourcesExt